react-components/src/architecture/concrete/observations/ObservationsView.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  CDF_TO_VIEWER_TRANSFORMATION,
+  Overlay3DCollection,
+  type OverlayInfo,
+  type CustomObjectIntersectInput,
+  type CustomObjectIntersection
+} from '@cognite/reveal';
+import { type DomainObjectIntersection } from '../../base/domainObjectsHelpers/DomainObjectIntersection';
+import { Changes } from '../../base/domainObjectsHelpers/Changes';
+import { type DomainObjectChange } from '../../base/domainObjectsHelpers/DomainObjectChange';
+import { type Observation, type ObservationIntersection, observationMarker } from './types';
+import { ClosestGeometryFinder } from '../../base/utilities/geometry/ClosestGeometryFinder';
+import { getColorFromStatus } from './color';
+
+type ObservationCollection = Overlay3DCollection<Observation>;
+
+export class ObservationsView extends GroupThreeView<ObservationsDomainObject> {
+  private readonly _overlayCollection: ObservationCollection = new Overlay3DCollection([]);
+
+  protected override calculateBoundingBox(): Box3 {
+    return this._overlayCollection
+      .getOverlays()
+      .reduce((box, overlay) => box.expandByPoint(overlay.getPosition()), new Box3());
+  }
+
+  protected override addChildren(): void {
+    const observations = this.domainObject.getObservations();
+
+    const selectedObservation = this.domainObject.getSelectedObservation();
+    const overlayInfos = createObservationOverlays(observations, selectedObservation);
+    this._overlayCollection.removeAllOverlays();
+    this._overlayCollection.addOverlays(overlayInfos);
+
+    this.addChild(this._overlayCollection);
+  }
+
+  public override update(change: DomainObjectChange): void {
+    super.update(change);
+
+    if (change.isChanged(Changes.geometry)) {
+      this.clearMemory();
+      this.invalidateRenderTarget();
+      this.invalidateBoundingBox();
+    } else if (change.isChanged(Changes.selected)) {
+      this.resetColors();
+      this.invalidateRenderTarget();
+    }
+  }
+
+  public override intersectIfCloser(
+    intersectInput: CustomObjectIntersectInput,
+    closestDistance: number | undefined
+  ): undefined | CustomObjectIntersection {
+    const { domainObject } = this;
+
+    const closestFinder = new ClosestGeometryFinder<DomainObjectIntersection>(
+      intersectInput.raycaster.ray.origin
+    );
+
+    if (closestDistance !== undefined) {
+      closestFinder.minDistance = closestDistance;
+    }
+
+    const intersectedOverlay = this._overlayCollection.intersectOverlays(
+      intersectInput.normalizedCoords,
+      intersectInput.camera
+    );
+
+    if (intersectedOverlay === undefined || !intersectedOverlay.getVisible()) {
+      return undefined;
+    }
+
+    const point = intersectedOverlay.getPosition();
+
+    if (domainObject.useClippingInIntersection && !intersectInput.isVisible(point)) {
+      return undefined;
+    }
+
+    if (!closestFinder.isClosest(point)) {
+      return undefined;
+    }
+
+    const customObjectIntersection: ObservationIntersection = {
```

Comment:
I will think this is overeengreneered and difficult to make a seperate ObservationIntersection.  You can stick with DomainObjectIntersection and skip the marker stuff. Can not see why you need this. The user data should be the data, not the overlay since this is a view specific thing. Use itersectedOverlay.getContent() instead.

Reply:
See above - I do this to be able to easily be able to tell which type each field in the intersection object has. If we want new observation intersection types, we can create separate types and symbols for those. Remember that the `as`-operator does not do anything in runtime, which means it's much more unsafe to use in Typescript than in e.g. C# - and creating this separate type allows us to avoid using it